client-spark/common/src/main/java/org/apache/spark/shuffle/reader/RssShuffleDataIterator.java

```diff
@@ -137,11 +139,9 @@ private int uncompress(CompressedShuffleBlock rawBlock, ByteBuffer rawData) {
 
     int uncompressedLen = rawBlock.getUncompressLength();
     if (codec != null) {
-      if (uncompressedData == null || uncompressedData.capacity() < uncompressedLen) {
-        // todo: support off-heap bytebuffer
-        uncompressedData = ByteBuffer.allocate(uncompressedLen);
-      }
-      uncompressedData.clear();
+      // todo: when we have netty data transportation, we will only use off heap memory.
+      uncompressedData = rawData.isDirect()
```

Comment:
Does this memory allocation need to be managed by the Spark memory management mechanism?

Reply:
One record won't occupy too much memory, we can hardly use off heap memory if we use Spark memory management. The Spark memory management usually provide heap memory.

Reply:
I'm concern about the high ratio compression of blocks for some apps.

Reply:
Anyway, currently it looks good to me.